tests/e2e/mcp_route_authorization_test.go

```diff
@@ -76,6 +81,37 @@ func TestMCPRouteAuthorization(t *testing.T) {
 		requireSumToolResult(ctx, t, sess, 41, 1, "42")
 	})
 
+	t.Run("allow rules with matching scopes and mismatching claims", func(t *testing.T) {
```

Comment:
nit: make the name not misleading
```suggestion
	t.Run("access denied with matching scopes and mismatching claims", func(t *testing.T) {
```